### ISSUE TITLE
Error: googleapi: Error 400: Must specify a non-empty Pub/Sub topic when Pub/Sub notification is enabled., badRequest

### ISSUE THREAD

**Issue by red8888** [terraform-google-modules/terraform-google-kubernetes-engine#1338]
### TL;DR

I tried to use the notification config like this:

```
  notification_config {
    pubsub {
      enabled = true
      topic = data.google_pubsub_topic.notifications.id
    }
  }
```

Got a super cryptic error I can find zero docs about: `Error: googleapi: Error 400: Must specify a non-empty Pub/Sub topic when Pub/Sub notification is enabled., badRequest`

### Expected behavior

For notifications to be configured without error

### Terraform Configuration

```hcl
notification_config {
    pubsub {
      enabled = true
      topic = data.google_pubsub_topic.notifications.id
    }
  }
```
```


### Terraform Version

```sh
Terraform v0.14.4
+ provider registry.terraform.io/hashicorp/google v3.72.0
+ provider registry.terraform.io/hashicorp/google-beta v3.72.0
```

### Additional information

When I look at the plan I see it appears to be totally ignoring the topic property:
```
      ~ notification_config {
          ~ pubsub {
              ~ enabled = false -> true
            }
        }
```

**Comment by red8888** [comment]
My bad. I had my data source misconfigured. 

Although I'm curious why topic is optional. seems like its required right?
